bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -0,0 +1,719 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel.blackboardutils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.lang3.StringUtils;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE;
+import org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.AccountFileInstance;
+import org.sleuthkit.datamodel.Blackboard;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.Relationship;
+import org.sleuthkit.datamodel.SleuthkitCase;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskDataException;
+
+/**
+ * A helper class to support modules that parse SQLite databases from mobile
+ * apps and create artifacts.
+ */
+public final class CommunicationArtifactsHelper extends ArtifactHelper {
+
+	private static final Logger logger = Logger.getLogger(CommunicationArtifactsHelper.class.getName());
+
+	/**
+	 * Enum for message read status
+	 */
+	public enum MessageReadStatus {
+
+		UNKNOWN, /// read status is unknown
+		UNREAD, /// message has not been read
+		READ        /// message has been read
+	}
+
+	/**
+	 * Enum for call/message direction
+	 */
+	public enum CommunicationDirection {
+		UNKNOWN("Unknown"),
+		INCOMING("Incoming"),
+		OUTGOING("Outgoing");
+
+		private final String dirStr;
+
+		CommunicationDirection(String dir) {
+			this.dirStr = dir;
+		}
+
+		public String getString() {
+			return dirStr;
+		}
+	}
+
```

Comment:
Why do two of these enums get display names and the other does not?

If we need display names, it is better to have a getDisplayName method. The getString method name used here is too close to the name of the toString method provided by the Enum base class.

Reply:
The MessageReadStatus eventually goes into an integer type attribute, which is why  strings weren't defined for it.  I can define the string but it would only be used for logging purposes or such.